core/src/main/scala/org/apache/spark/scheduler/cluster/SparkDeploySchedulerBackend.scala

```diff
@@ -78,7 +78,11 @@ private[spark] class SparkDeploySchedulerBackend(
     val appUIAddress = sc.ui.map(_.appUIAddress).getOrElse("")
     val appDesc = new ApplicationDescription(sc.appName, maxCores, sc.executorMemory, command,
       appUIAddress, sc.eventLogDir)
-
+    appDesc.coreNumPerTask = conf.getInt("spark.task.cpus", 1)
```

Comment:
I chose var with the concern that, if I use val, that means I need to add it as the constructor parameter...just not sure if I can pass the binary compatibility checking